### PR TITLE
Added the Get File Sample - Generic playbook

### PR DESCRIPTION
**Pull request by idovandijk**
## Status
~In Progress - Merge only after https://github.com/demisto/content/pull/24486~
Ready

## Related Issues
Fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5582
Related to: Related to: https://jira-hq.paloaltonetworks.local/browse/CIAC-4156

## Description
Created a new playbook to retrieve files from endpoints either by path or by hash, and optionally unzip them as well.

## Screenshots
Get_File_Sample_-_Generic

## Minimum version of Cortex XSOAR
6.5.0

## Does it break backward compatibility?
No

### CODE REVIEW

**Review by tomer-pan — APPROVED** [review]

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    id: "1"
+    taskid: 7456030a-675f-4c46-8231-7820167addc6
+    type: playbook
+    task:
+      id: 7456030a-675f-4c46-8231-7820167addc6
+      version: -1
+      name: Get File Sample By Hash - Generic v3
+      description: |-
+        This playbook returns a file sample correlating to a hash in the War Room using the following sub-playbooks:
+        - Get binary file by MD5 hash from Carbon Black telemetry data - VMware Carbon Black EDR v2.
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:48]:
```suggestion
        - VMware Carbon Black EDR v2 - Get binary file by MD5 hash from Carbon Black telemetry data.
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+        - Get the threat (file) attached to a specific SHA256 hash - Cylance Protect v2.
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:49]:
```suggestion
        - Cylance Protect v2 - Get the threat (file) attached to a specific SHA256 hash .
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml (hunk 1 above)
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:48]: Is this the name of the sub-playbook?

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml (hunk 2 above)
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:49]: Is this the name of the sub-playbook?

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+    id: "2"
+    taskid: 5759c2d5-753c-4591-86f6-ff7d4647f306
+    type: playbook
+    task:
+      id: 5759c2d5-753c-4591-86f6-ff7d4647f306
+      version: -1
+      name: Get File Sample From Path - Generic V3
+      description: |
+        This playbook returns a file sample from a specified path and host that you input in the following playbooks:
+        1) PS Remote Get File Sample From Path.
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:98]: Is this the name of the playbook?

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+        2) Get File Sample From Path - VMware Carbon Black EDR (Live Response API).
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:99]: Is this the correct name of the sub-playbook?

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+  required: false
+  description: The SHA256 hash of the file to retrieve.
+  playbookInputQuery:
+- key: Hostname
+  value:
+    complex:
+      root: Endpoint
+      accessor: Hostname
+  required: false
+  description: The hostname from which to retrieve the f ile.
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:296]:
```suggestion
  description: The hostname from which to retrieve the file.
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+  playbookInputQuery:
+- key: AgentID
+  value:
+    complex:
+      root: incident
+      accessor: agentid
+  required: false
+  description: The Agent ID that should be used to retrieve the file.
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:304]:
```suggestion
  description: The agent ID that should be used to retrieve the file.
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+  playbookInputQuery:
+- key: FilePath
+  value:
+    complex:
+      root: incident
+      accessor: filepath
+  required: false
+  description: "The full path to the file that needs to be retrieved."
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:312]:
```suggestion
  description: "he full path to the file that needs to be retrieved.
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+  description: 'The tool used to unzip the file. The options are: 7z, zipfile.'
+  playbookInputQuery:
+- key: ZipPassword
+  value: {}
+  required: false
+  description: The password used to unzip zipped files.
+  playbookInputQuery:
+outputs:
+- contextPath: File
+  description: File objects - includes the zipped file and any unzipped files.
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:336]:
```suggestion
  description: File objects. Includes the zipped file and any unzipped files.
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml [hunk 9]
```diff
@@ -0,0 +1,377 @@
...
+  type: unknown
+- contextPath: File.Size
+  description: The size of the file.
+  type: number
+- contextPath: File.Type
+  description: The type of the file.
+  type: string
+- contextPath: File.Info
+  description: General information of the file.
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:345]:
```suggestion
  description: General information about the file.
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+This playbook does not use any commands.
+
+## Playbook Inputs
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| MD5 | The MD5 hash of the file to retrieve. | File.MD5 | Optional |
+| SHA256 | The SHA256 hash of the file to retrieve. | File.SHA256 | Optional |
+| Hostname | The hostname from which to retrieve the f ile. | Endpoint.Hostname | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md:27]:
```suggestion
| Hostname | The hostname from which to retrieve the file. | Endpoint.Hostname | Optional |
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+| AgentID | The Agent ID that should be used to retrieve the file. | incident.agentid | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md:28]:
```suggestion
| AgentID | The agent ID that should be used to retrieve the file. | incident.agentid | Optional |
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+| FilePath |  | incident.filepath | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md:29]:
```suggestion
| FilePath | The full path to the file that needs to be retrieved. | incident.filepath | Optional |
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+| UnzipFile | Whether to unzip zipped files after retrieving them. |  | Optional |
+| ZipTool | The tool used to unzip the file. The options are: 7z, zipfile. |  | Optional |
+| ZipPassword | The password used to unzip zipped files. |  | Optional |
+
+## Playbook Outputs
+---
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| File | File objects - includes the zipped file and any unzipped files. | unknown |
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md:40]:
```suggestion
| File | File objects. Includes the zipped file and any unzipped files. | unknown |
```

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| MD5 | The MD5 hash of the file to retrieve. | File.MD5 | Optional |
+| SHA256 | The SHA256 hash of the file to retrieve. | File.SHA256 | Optional |
+| Hostname | The hostname from which to retrieve the f ile. | Endpoint.Hostname | Optional |
+| AgentID | The Agent ID that should be used to retrieve the file. | incident.agentid | Optional |
+| FilePath |  | incident.filepath | Optional |
+| NewFilename | Optional - a new name for the retrieved file. If left empty, the filename will not change. |  | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic_README.md:30]:
```suggestion
| NewFilename | Optional. A new name for the retrieved file. If left empty, the filename will not change. |  | Optional |
```

File: Packs/CommonPlaybooks/ReleaseNotes/2_3_35.md [hunk 15]
```diff
@@ -0,0 +1,4 @@
+
+#### Playbooks
+##### New: Get File Sample - Generic
+- Retrieve files from endpoints by the file hash or the file path. (Available from Cortex XSOAR 6.5.0).
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/ReleaseNotes/2_3_35.md:4]:
```suggestion
Retrieve files from endpoints by the file hash or the file path. (Available from Cortex XSOAR 6.5.0).
```

**Comment by ShirleyDenkberg** [comment]
@michalgold @tomer-pan Doc review completed.

File: Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml (hunk 1 above)
Comment (idovandijk) [reply Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:48]: Yes that's its name

Comment (idovandijk) [reply Packs/CommonPlaybooks/Playbooks/Get_File_Sample_-_Generic.yml:48]: This is a description coming from another playbook I did not touch. But the correct subplaybooks of it are:
`Get the binary file from Carbon Black by its MD5 hash`
`Get File Sample By Hash - Cylance Protect v2`
`Code42 File Download`

**Pull request merged**